ovos_PHAL_plugin_homeassistant/__init__.py

```diff
@@ -752,55 +751,103 @@ def handle_set_group_display_settings(self, message):
             self.handle_show_dashboard()
 
 # OAuth QR Code Flow Handlers
-    def request_host_info_from_oauth(self):
-        self.bus.emit(Message("oauth.get.app.host.info"))
+    def request_host_info_from_oauth(self, message):
+        """
+        Get the oauth server configuration for this device
+        @param message: Message associated with oauth start request
+        @return:
+        """
+        message = message.forward("oauth.get.app.host.info")
+        resp = self.bus.wait_for_response(message,
+                                          "oauth.app.host.info.response")
+        if not resp:
+            raise RuntimeError(f"No response from oauth plugin to message: "
+                               f"{message.msg_type}: {message.data}")
+        self.handle_oauth_host_info(resp)
 
     def handle_oauth_host_info(self, message):
+        """
+        Handle a response message with oauth host/port for this device
+        @param message: oauth.app.host.info.response Message from oauth plugin
+        """
         host = message.data.get("host", None)
         port = message.data.get("port", None)
         self.oauth_client_id = f"http://{host}:{port}"
-
+        LOG.info(f"Got oauth endpoint: {self.oauth_client_id}")
         if self.temporary_instance:
             self.oauth_register()
-            self.start_oauth_flow()
+            self.start_oauth_flow(message)
+        else:
+            LOG.error(f"Unexpected oauth message: {message.msg_type}")
 
     def handle_start_oauth_flow(self, message):
-        """ Handle the start oauth flow message
+        """
+        Handle a request to start oauth login, i.e. from a GUI page
 
             Args:
                 message (Message): The message object
         """
-        instance = message.data.get("instance", None)
+        instance = message.data.get("instance").lower()
```

Comment:
`None` will not have method `lower()`, so we'll need a default empty string. That will still evaluate as falsey in the line below and won't potentially throw an exception.